### PR TITLE
terraforming: init at 0.18.0

### LINKED ISSUE #35673: Package request: Terraforming
## Issue description
Please package https://github.com/dtan4/terraforming.

### PR DESCRIPTION
**Pull request by kalbasit**
###### Motivation for this change

Introduce [terraforming](https://github.com/dtan4/terraforming) at its latest version.

closes #35673

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -29925,6 +29925,8 @@ in
     callPackage ../applications/networking/cluster/terraform-providers {}
   );
 
+  terraforming = callPackage ../applications/networking/cluster/terraforming {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:29928]:
```suggestion
  terraforming = callPackage ../applications/networking/cluster/terraforming { };
```

File: pkgs/applications/networking/cluster/terraforming/Gemfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+source "https://rubygems.org"
+
+gem 'terraforming'
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/terraforming/Gemfile:3]: Ideally we would pin this gem to make the lockfile more reproducible.

Comment (kalbasit) [reply pkgs/applications/networking/cluster/terraforming/Gemfile:3]: I don't think that's necessary, in fact, I think it'll do more harm than good. The Gemfile.lock/gemset.nix already lock to a specific version and whenever I run the updater script I expect to just grab the latest and greatest for me to test and submit as PR; Having to explicitly have to edit the Gemfile defeats the whole point of the updaterScript. The only useful usecase I can see for this is to roll out a bug/security fix for a specific pinned version which we can do temporarily at the time of that update.

**Comment by kalbasit** [comment]
Failing for me on Darwin 🤔 

```
$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /Users/yl/.cache/nixpkgs-review/pr-117731-2/build.nix
error: getting attributes of path '/usr/lib/libSystem.B.dylib: No such file or directory
(use '--show-trace' to show detailed location information)

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/117731

1 package failed to build:
terraforming
```

**Comment by kalbasit** [comment]
Interesting. It failed when I ran it via nix-review but it worked when I did it with `nix-build`. I'm using a pristine Mac that does not have any configurations beyond the initial single-user installation.

**Comment by SuperSandro2000** [comment]
> Interesting. It failed when I ran it via nix-review but it worked when I did it with `nix-build`. I'm using a pristine Mac that does not have any configurations beyond the initial single-user installation.

nixpkgs-review turns on sandbox. This is disabled on darwin for the next release.

**Pull request merged**